Index: Makefile
===================================================================
RCS file: /cvs/xenocara/lib/Makefile,v
retrieving revision 1.46
diff -u -p -u -p -r1.46 Makefile
--- Makefile	11 Nov 2021 08:59:43 -0000	1.46
+++ Makefile	1 Aug 2023 18:42:06 -0000
@@ -51,6 +51,7 @@ SUBDIR+= libxshmfence
 
 .if ${XENOCARA_BUILD_GL:L} == "yes"
 SUBDIR+= libdrm
+SUBDIR+= libva
 SUBDIR+= mesa
 SUBDIR+= libGLU libGLw
 SUBDIR+= libepoxy
